_includes/comments-providers/disqus.html

```diff
@@ -0,0 +1,22 @@
+{%- if page.key and
+  site.comments.disqus.shortname -%}
+
+  <div id="disqus_thread"></div>
+  <script>
+  /**
+  *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
+  *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
+  var disqus_config = function () {
+    this.page.url = '{%- include snippets/page-url.html -%}';
+    this.page.identifier = '{{ page.key }}';
+  };
+  (function() { // DON'T EDIT BELOW THIS LINE
+    var d = document, s = d.createElement('script');
+    s.src = 'https://{{ site.comments.disqus.shortname }}.disqus.com/embed.js';
+    s.setAttribute('data-timestamp', +new Date());
+    (d.head || d.body).appendChild(s);
+  })();
+  </script>
+  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
```

Comment:
Here is the previous version:

```html

  <hr>
  <div id="disqus_thread"></div>
  <script>
    var disqus_config = function () {
      this.page.url = "{{ site.url }}{{ page.url }}";
      this.page.identifier = "{{ page.id }}";
    };
    (function() { // DON'T EDIT BELOW THIS LINE
      var d = document, s = d.createElement('script');
      s.src = 'https://mincong-h.disqus.com/embed.js';
      s.setAttribute('data-timestamp', +new Date());
      (d.head || d.body).appendChild(s);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
```

The changes are:

- [ ] Page URL (`page.url`)
- [ ] Page ID (`page.identifier`)
- [x] Disqus shortname

Reply:
Maybe it's normal that no comments in shown in localhost because the comment section is disabled in development environment:

```rb
{%- if jekyll.environment != 'development' -%}
  {%- if site.comments.provider == 'disqus' -%}
    {%- include comments-providers/disqus.html -%}
  {%- elsif site.comments.provider == 'gitalk' -%}
    {%- include comments-providers/gitalk.html -%}
  {%- elsif site.comments.provider == 'valine' -%}
    {%- include comments-providers/valine.html -%}
  {%- elsif site.comments.provider == 'custom' -%}
    {%- include comments-providers/custom.html -%}
  {%- endif -%}
{%- endif -%}
```

Reply:
Validate the changes by temporarily enabling the production mode (`JEKYLL_ENV=production`). We can see that the comments are correctly loaded:

![Screenshot 2021-02-12 at 19 23 42](https://user-images.githubusercontent.com/10179217/107807018-de166200-6d67-11eb-8c77-30fb03ae39cd.png)